sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -275,10 +275,11 @@ shr(T1 src0, T2 src1, Sat sat = {}) {
 /// the input vector \p src0 shall be rotated.
 /// @return vector of rotated elements.
 template <typename T0, typename T1, int SZ>
-__ESIMD_API
-    std::enable_if_t<std::is_integral<T0>::value && std::is_integral<T1>::value,
-                     __ESIMD_NS::simd<T0, SZ>>
-    rol(__ESIMD_NS::simd<T1, SZ> src0, __ESIMD_NS::simd<T1, SZ> src1) {
+__ESIMD_API std::enable_if_t<
+    __ESIMD_NS::detail::is_type<T0, int16_t, uint16_t, int32_t, uint32_t>() &&
```

Comment:
This code needs more corrections to follow the restrictions in HW instructions.
1) int64 vectors are allowed for PVC+

2) it is not obvious from GenX-Intrinsics and bSPEC, but the HW instruction expects same bit-width for destination and source. If so, we could pass only 1 type to __esimd_rol

Potentially, instead of generating __esimd_rol we could generate SPIRV instruction rotate from this extended inst set: https://registry.khronos.org/SPIR-V/specs/unified1/OpenCL.ExtendedInstructionSet.100.html
Not sure though if GPU RT already supports it.
If it is supported, then perhaps it could emulate rol/ror for int64 even when HW doesn't support it.

Adding @vmustya and @kbobrovs to reviewers as well